crates/weaver_resolver/src/registry.rs

```diff
@@ -800,6 +764,94 @@ mod tests {
         }
     }
 
+    fn create_registry_from_string(registry_spec: &str) -> Result<Registry, crate::Error> {
```

Comment:
IMO, the right way to do this will be:
- Create a `SemConvRegistry` with the `new` constructor.
- Add a new method to `SemConvRegistry` to load a semantic convention group similar to `load_from_file` but for `&str`. This method could be named `load_from_string` and take a `&str` as a parameter.
- Remove the dependency to tempfile

Reply:
Makes sense, I added a `load_from_str` method to the `SemConvRegistry`.